articles/adding-dark-theme-to-your-site/index.md

```diff
@@ -0,0 +1,178 @@
+    --color-heading: #0077ff; 
+    --color-text: #B5B5B5;
+}
+```
+
+Next, we need to specify the CSS variables' use through the CSS file so that all the website elements will change when we click the toggle theme button.
+
+```css
+body { 
+    background-color: var(--bg);
+}
+
+.container {
+    background-color: var(--bg-panel);
+
+    h1{
+        margin: 0;
+        color: var(--color-heading)
+    }
+
+    p{
+        color: var(--color-text);
+       
+    }
+
+}
+```
+
+### Switch Theme Button
+In the next section, we will style our toggle switch which will be cheched to switch bewten the dark and ligh theme. The code for this toggle button cis as below. You can find the  code to this switch [here](https://codepen.io/mburnette/pen/LxNxNg)
+
+```css
+    input[type=checkbox] {
+        height: 0;
+        width: 0;
+        visibility: hidden;
+    }
+
+    label {
+        cursor: pointer;
+        text-indent: -9999px;
+        width: 52px;
+        height: 27px;
+        background: #1ba80e;
+        float: right;
+        border-radius: 100px;
+        position: relative;
+    }
+
+    label:after {
+        content: '';
+        position: absolute;
+        top: 3px;
+        left: 3px;
+        width: 20px;
+        height: 20px;
+        background: #fff;
+        border-radius: 90px;
+        -webkit-transition: 0.3s;
+        transition: 0.3s;
+    }
+
+    input:checked + label {
+        background: var(--color-heading);
+    }
+
+    input:checked + label:after {
+        left: calc(100% - 5px);
+        -webkit-transform: translateX(-100%);
+                transform: translateX(-100%);
+    }
+
+    label:active:after {
+        width: 45px;
+    }
+```
+
+### Adding the JavaScript
+We will handle the theme switching through JavaScript by changing the toggle switch's class name to either light or dark, as defined above. In doing so, we write two functions `changeThemeToDark()` and `chnageThemeToWhite()`.
+
+**Change the theme to dark.**
+```javascript
+    // Change theme to dark by adding the `dark` classname to html element.
+   const changeThemeToDark = () =>{
+    document.documentElement.setAttribute("data-theme", "dark")
+}
+
+```
+
+**Change the theme to Light.**
+```javascript
+    // Reset the html class to default
+   const changeThemeToDark = () =>{
+    document.documentElement.setAttribute("data-theme", "dark");
+}
+```
+
+### Using Local Storage to remember the theme.
+Now that our dark theme is set, we need to enable the browser to remember the theme our user chose so that in subsequent visits or page refreshing, they see their preferred theme. We can use local storage to implement this functionality. We will store the applied theme in local storage and read subsequent calls of the `changeThemeToDark()` and `chnageThemeToWhite().` Even after refreshing the page, the theme is still reserved. We are using local storage because the data stored in local storage has no expiration time. The 
```

Comment:
```suggestion
Now that our dark theme is set, we need to enable the browser to remember the theme our user chose, so that in subsequent visits or page refreshing, they see their preferred theme.

We can use local storage to implement this functionality. We will store the applied theme in local storage and read subsequent calls of the `changeThemeToDark()` and `changeThemeToWhite().` Even after refreshing the page, the theme is still reserved. We are using local storage because the data stored in local storage has no expiration time.
```